NuRadioReco/modules/RNO_G/stationLDAVariables.py

```diff
@@ -0,0 +1,229 @@
+import logging
+
+import matplotlib
+matplotlib.rcParams.update({"font.size": 20})
+import numpy as np
+import matplotlib.pyplot as plt
+from collections import defaultdict
+from fnmatch import fnmatch
+import pymap3d as pm
+import codecs, json
+from math import ceil
+import math
+from scipy import signal, stats
+import scipy.optimize as opt
+import json
+import mattak.Dataset as mattak
+from scipy.signal import hilbert, butter, lfilter, resample
+#import global_constants
+import uproot
+from NuRadioReco.utilities import units, fft
+from NuRadioReco.modules.RNO_G.channelBlockOffsetFitter import fit_block_offsets
+from scipy.ndimage import maximum_filter1d, minimum_filter1d
+from NuRadioReco.modules.base.module import register_run
+
+
+class stationLDAVariables:
+    """
+    Module that calculates some variables for a Linear Discriminant Analysis of a specific event and stores them at the station level.
+
+
+    """
+
+    def __init__(self):
+        self.logger = logging.getLogger('NuRadioReco.stationLDAVariables')
+
+    def begin(self,coincidence_window_size=6,pad_length=500,channel_ids=[0,1,2,3]):
+        """
+        Parameters
+        ----------
+
+        coincidence_window_size : window size used for calculating the maximum peak to peak amplitude
+
+        pad_length : padding length used for calculating the coherent sum
+
+        channel_ids : channels for which to calculate the variables
+
+        """
+        self.sum_chan=None
+        self.__coincidence_window_size=coincidence_window_size
+        self.__pad_length=pad_length
+        self.__channel_ids=channel_ids
+
+
+    @register_run()
+    def run(self, event, station, detector, channel_id=0):
+
+        """
+        Calculate LDA variables and add to the station object.
+
+        Parameters
+        ----------
+
+        event
+
+        station
+
+        detector
+
+        """
+        # print('channel:', station.get_channel(channel_id))
+        # print('max_a=',self.max_a(event, station, detector))
+        # print('avg_ch_snr=',self.avg_ch_snr(event, station, detector))
+        # print('coherent_sum_step_by_step=',self.coherent_sum_step_by_step(event, station))
+        # print('coherent_sum=',self.coherent_sum(event, station, station.get_channel(channel_id)))
+        # print('coherent_snr=',self.coherent_snr(self.sum_chan))
+        # print('impulsive_value=',self.impulsive_value(station.get_channel(channel_id).get_trace()))
+        station.set_parameter('max_a', self.max_a(event, station, detector))
+        station.set_parameter('avg_ch_snr', self.avg_ch_snr(event, station, detector))
+        self.coherent_sum(event, station, station.get_channel(channel_id))
+        station.set_parameter('coherent_snr', self.coherent_snr(event, station, detector))
```

Comment:
passing strings as first argument to `set_parameter` will not work. You have to use parameter:
```
from NuRadioReco.framework.parameters import channelParameters as chp, stationParameter as stp
...
station.set_parameter(stp.NAME, value)
```